node-manager/src/node.rs

```diff
@@ -29,4 +32,129 @@ impl Node {
             keys_manager: Arc::new(keys_manager),
         })
     }
+
+    pub async fn connect_peer(
+        &self,
+        websocket_proxy_addr: String,
+        peer_pubkey_and_ip_addr: String,
+    ) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
this can be a MutinyError or could create a new sub-error like MutinyConnectionError or something for these

the shortcut is to just to return `anyhow::Result<()>` (I like to be explicit that I'm using the anyhow result, because it's a little magic and hides the error type, and will eventually want to swap out with a more concrete error type)

then when you want an error use anyhow's `bail!` macro to return early, or use `?` to return another lib's errors

in the long run it's nice to write all our own errors but also don't want to block on it so anyhow is a nice shortcut when you're not in the mood

Reply:
if you want to return an error from another lib, but don't want to totally erase it by coercing it into some generic anyhow error, you can do `.with_context(...)`

Reply:
Changed to `MutinyError`.